python/ray/serve/_private/proxy_state.py

```diff
@@ -190,107 +188,98 @@ def actor_handle(self) -> ActorHandle:
         """
         return self._actor_handle
 
-    @property
-    def health_check_ongoing(self) -> bool:
-        """Return whether the health check is ongoing or not."""
-        return self._health_check_obj_ref is not None
-
-    @property
-    def is_draining(self) -> bool:
-        """Return whether the drained check is ongoing or not."""
-        return self._is_drained_obj_ref is not None
-
-    def reset_drained_check(self):
-        """Reset the drained check object reference."""
-        self._is_drained_obj_ref = None
-
-    def reset_health_check(self):
-        """Reset the health check object reference."""
-        self._health_check_obj_ref = None
-
-    def start_new_ready_check(self):
-        """Start a new ready check on the proxy actor."""
-        self._ready_obj_ref = self._actor_handle.ready.remote()
-
-    def start_new_health_check(self):
-        """Start a new health check on the proxy actor."""
-        self._health_check_obj_ref = self._actor_handle.check_health.remote()
+    def is_ready(self, timeout_s: float) -> Optional[bool]:
+        if self._ready_check_future is None:
+            self._ready_check_future = wrap_as_future(
+                self._actor_handle.ready.remote(), timeout_s=timeout_s
+            )
```

Comment:
Ha! That's actually what it's been before, but @GeneDer made a good point on just letting it fall through so that if future has failed right away we don't wait for another iteration

https://github.com/ray-project/ray/pull/41722/files#r1421100067

Reply:
I think it's this thread https://github.com/ray-project/ray/pull/41722#discussion_r1421097266 But I don't have strong preference for either 😛